File: sdks/ruby/lib/novabilling/types/invoice_item_dto.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class InvoiceItemDto < Internal::Types::Model
      field :description, -> { String }, optional: false, nullable: false
      field :quantity, -> { Integer }, optional: false, nullable: false
      field :unit_amount, -> { Integer }, optional: false, nullable: false, api_name: "unitAmount"
    end
  end
end

File: sdks/csharp/src/NovaBilling/Types/InvoiceItemDto.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record InvoiceItemDto : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("description")]
    public required string Description { get; set; }

    [JsonPropertyName("quantity")]
    public required double Quantity { get; set; }

    [JsonPropertyName("unitAmount")]
    public required double UnitAmount { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
